ci: Mono repo setup

frascuchon opened the pull request:
The mono repo structure will divide the argilla repository into different folders, 
each one containing a different part of the project. 

The `.github/workflows` folder will contain the different workflows for each part of the project because of the
limitation of GitHub Actions to have only one workflow folder per repository.

The `docs` (or `argilla-docs`) folder will contain the documentation for the whole project.

The rest of the folders will contain the different parts of the project, such as the SDK, 
the frontend, and the server.

Each part of the project will have its own `CHANGELOG.md` file and its own `README.md` file.

Contribution guidelines will be in the `CONTRIBUTING.md` file and the code of conduct will be in the `CODE_OF_CONDUCT.md` file.

A example of the project structure would be:

```
- .github/workflows/
    |_ argilla.yml # test + build argilla
    |_ argilla-frontend.yml # test + build argilla-frontend
    |_ argilla-server.yml # test + build argilla-server
    ...
- argilla-sdk/ #  -> V2
    |_ CHANGELOG.md
    |_ README.md
    |_ src/
    |_ tests/
    ...
- argilla/ # -> V1
    |_ CHANGELOG.md
    |_ README.md
    |_ src/
    |_ tests/
    ...
- argilla-frontend/
    |_ CHANGELOG.md
    |_ README.md
    |_ package.json
    |_ nuxt.config.ts
	...
- argilla-server/
    |_ CHANGELOG.md
    |_ README.md
    |_ src/
    |_ tests/
    ...
- docs/
    ...
- CONTRIBUTING.md
- CODE_OF_CONDUCT.md
...
```

frascuchon committed:
ci: Create a sample test to run action under folder

frascuchon committed:
ci: Move sample workflow to parent folder

frascuchon committed:
create argilla folder

frascuchon committed:
create argilla-frontend folder

frascuchon committed:
ci: Configure workflows with new paths

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

frascuchon committed:
chore: Change script

frascuchon committed:
Merge branch 'ci/configure-mono-repo' of github.com:argilla-io/argilla into ci/configure-mono-repo

frascuchon committed:
ci: Clean and organize workflows

frascuchon committed:
Merge branch 'develop' into ci/configure-mono-repo

frascuchon committed:
ci: Organize workflows at workflows folder

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

frascuchon committed:
ci: Configure main workflow for `argilla-frontend` (#4760)

<!-- Thanks for your contribution! As part of our Community Growers
initiative 🌱, we're donating Justdiggit bunds in your name to reforest
sub-Saharan Africa. To claim your Community Growers certificate, please
contact David Berenstein in our Slack community or fill in this form
https://tally.so/r/n9XrxK once your PR has been merged. -->

# Description

This PR adds a basic workflow to build and deploy the `argilla-frontend`
project. The workflow tests and builds frontend resources, uploads
static distribution as an artifact, builds a dev docker image and
deploys it in an isolated environment.



**Type of change**

(Please delete options that are not relevant. Remember to title the PR
according to the type of change)

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (change restructuring the codebase without changing
functionality)
- [ ] Improvement (change adding some improvement to an existing
functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And
ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK)
(see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See
https://keepachangelog.com/)

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: José Francisco Calvo <jose@argilla.io>

frascuchon committed:
Merge branch 'develop' into ci/configure-mono-repo

frascuchon committed:
chore: Remove file copies

frascuchon committed:
chore: Restore changes from default branch

frascuchon committed:
ci: Configure `argilla-server` project (#4765)

<!-- Thanks for your contribution! As part of our Community Growers
initiative 🌱, we're donating Justdiggit bunds in your name to reforest
sub-Saharan Africa. To claim your Community Growers certificate, please
contact David Berenstein in our Slack community or fill in this form
https://tally.so/r/n9XrxK once your PR has been merged. -->

# Description

This PR defines a workflow pipeline to build the `argilla-server`
package and the related docker images. The `argilla-server` folder will
be included as a separate PR once the
https://github.com/argilla-io/argilla-server is combined here. The
existing folder (if any) will be used for test purposes.

See a successful run here:
https://github.com/argilla-io/argilla/actions/runs/8968610122

**Type of change**

(Please delete options that are not relevant. Remember to title the PR
according to the type of change)

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (change restructuring the codebase without changing
functionality)
- [X] Improvement (change adding some improvement to an existing
functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And
ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK)
(see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See
https://keepachangelog.com/)

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

frascuchon committed:
chore: Temporal argilla-frontend folder

frascuchon committed:
chore: Temporal argilla folder

frascuchon committed:
chore: Temporal argilla-server folder

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks
for more information, see https://pre-commit.ci

frascuchon committed:
ci: Align events definitions

frascuchon committed:
Merge branch 'ci/configure-mono-repo' of github.com:argilla-io/argilla into ci/configure-mono-repo

frascuchon committed:
fix: reusable path

frascuchon committed:
Replace `'` by `"`

frascuchon committed:
chore: format argilla frontend workflow

frascuchon committed:
ci: Review the `argilla` workflow  (#4783)

<!-- Thanks for your contribution! As part of our Community Growers
initiative 🌱, we're donating Justdiggit bunds in your name to reforest
sub-Saharan Africa. To claim your Community Growers certificate, please
contact David Berenstein in our Slack community or fill in this form
https://tally.so/r/n9XrxK once your PR has been merged. -->

# Description

This PR reviews and changes the `argilla.yml` workflow and makes
everything work.

- Some of the action versions have been upgraded.
- The check repo files workflow has been removed.
- The paths relative to `argilla` have been reviewed and updated.



**Type of change**

(Please delete options that are not relevant. Remember to title the PR
according to the type of change)

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (change restructuring the codebase without changing
functionality)
- [x] Improvement (change adding some improvement to an existing
functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And
ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK)
(see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See
https://keepachangelog.com/)

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

frascuchon committed:
remove new project files

frascuchon committed:
Merge branch 'develop' into ci/configure-mono-repo

frascuchon committed:
restore files from default branch

frascuchon committed:
restore file

jfcalvo commented on .github/workflows/argilla-server.yml:
```diff
@@ -0,0 +1,192 @@
+
+  release:
+    types:
+      - "published"
```
```suggestion
      - published
```

jfcalvo commented on .github/workflows/argilla-server.yml:
```diff
@@ -0,0 +1,192 @@
+      - name: Install dependencies
+        run: pdm install
+
+      - name: Run tests 📈
```
We're running tests only with PostgreSQL?

frascuchon replied:
Yes. I will include a separate step to run tests with SQLite (and other python engines) later. 

I decided to keep this first version quite simple, without matrix configurations.

frascuchon committed:
restore more changes from develop

frascuchon committed:
Apply suggestions from code review

Co-authored-by: José Francisco Calvo <jose@argilla.io>

frascuchon committed:
ci: moving argilla project (#4816)

<!-- Thanks for your contribution! As part of our Community Growers
initiative 🌱, we're donating Justdiggit bunds in your name to reforest
sub-Saharan Africa. To claim your Community Growers certificate, please
contact David Berenstein in our Slack community or fill in this form
https://tally.so/r/n9XrxK once your PR has been merged. -->

# Description

Moving the argilla project to `argilla` folder. 



**Type of change**

(Please delete options that are not relevant. Remember to title the PR
according to the type of change)

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (change restructuring the codebase without changing
functionality)
- [ ] Improvement (change adding some improvement to an existing
functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And
ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK)
(see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See
https://keepachangelog.com/)

frascuchon committed:
ci: moving argilla-frontend project (#4815)

<!-- Thanks for your contribution! As part of our Community Growers
initiative 🌱, we're donating Justdiggit bunds in your name to reforest
sub-Saharan Africa. To claim your Community Growers certificate, please
contact David Berenstein in our Slack community or fill in this form
https://tally.so/r/n9XrxK once your PR has been merged. -->

# Description

This PR moves files from `frontend` to `argilla-frontend`

**Type of change**

(Please delete options that are not relevant. Remember to title the PR
according to the type of change)

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (change restructuring the codebase without changing
functionality)
- [ ] Improvement (change adding some improvement to an existing
functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And
ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK)
(see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See
https://keepachangelog.com/)

jfcalvo committed:
ci: add argilla-server project (#4824)

# Description

Add `argilla-server` to his own folder using code from
https://github.com/argilla-io/argilla-server/pull/168

The following code has been removed/changed:
* Folder `/.github` has not been copied from `argilla-server`.
* Submodule `argilla` has not been copied from `argilla-server`.

**Type of change**

(Please delete options that are not relevant. Remember to title the PR
according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to not work as expected)
- [x] Refactor (change restructuring the codebase without changing
functionality)
- [ ] Improvement (change adding some improvement to an existing
functionality)
- [ ] Documentation update

**How Has This Been Tested**

- [ ] Build should be successfully created.

**Checklist**

- [ ] I added relevant documentation
- [ ] follows the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK)
(see text above)
- [ ] I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

jfcalvo committed:
ci: check argilla actions using argilla-server (#4826)

# Description

We will use this PR to check that GitHub Actions are working as expected
for `argilla` folder, integrated with new `argilla-server` project.

**Type of change**

(Please delete options that are not relevant. Remember to title the PR
according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing
functionality)
- [ ] Improvement (change adding some improvement to an existing
functionality)
- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And
ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [ ] follows the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK)
(see text above)
- [ ] I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

frascuchon committed:
chore: Delete old workflows

frascuchon committed:
Merge branch 'develop' into ci/configure-mono-repo

frascuchon committed:
chore: move deployment setup to examples (#4830)

<!-- Thanks for your contribution! As part of our Community Growers
initiative 🌱, we're donating Justdiggit bunds in your name to reforest
sub-Saharan Africa. To claim your Community Growers certificate, please
contact David Berenstein in our Slack community or fill in this form
https://tally.so/r/n9XrxK once your PR has been merged. -->

# Description

This PR moves files related to deployment setup into an
`examples/deployments` folder.

**Type of change**

(Please delete options that are not relevant. Remember to title the PR
according to the type of change)

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (change restructuring the codebase without changing
functionality)
- [x] Improvement (change adding some improvement to an existing
functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And
ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK)
(see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See
https://keepachangelog.com/)

---------

Co-authored-by: José Francisco Calvo <jose@argilla.io>

frascuchon committed:
Merge branch 'develop' into ci/configure-mono-repo

frascuchon committed:
Merge branch 'ci/configure-mono-repo' of github.com:argilla-io/argilla into ci/configure-mono-repo

frascuchon committed:
docs: Review Developer docs (#4776)

<!-- Thanks for your contribution! As part of our Community Growers
initiative 🌱, we're donating Justdiggit bunds in your name to reforest
sub-Saharan Africa. To claim your Community Growers certificate, please
contact David Berenstein in our Slack community or fill in this form
https://tally.so/r/n9XrxK once your PR has been merged. -->

# Description

This PR reviews the development docs related to this new mono repo
structure. I realized that some of those docs were outdated. These docs
will be revisited in future PRs.

**Type of change**

(Remember to title the PR according to the type of change)

- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes.)

- [ ] `sphinx-autobuild` (read [Developer
Documentation](https://docs.argilla.io/en/latest/community/developer_docs.html#building-the-documentation)
for more details)

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK)
(see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See
https://keepachangelog.com/)

---------

Co-authored-by: Natalia Elvira <126158523+nataliaElv@users.noreply.github.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

jfcalvo committed:
chore: upgrade argilla-frontend folder with latests changes from frontend folder at develop branch

jfcalvo committed:
Merge branch 'develop' into ci/configure-mono-repo
